dts/bindings/memory-controllers/atmel,sam-smc.yaml

```diff
@@ -0,0 +1,129 @@
+  SRAM, PSRAM, PROM, EPROM, EEPROM, LCD Module, NOR Flash and NAND Flash.
+
+  The SMC controller can have up to 4 children defining the connected external
+  memory devices. Device Tree example:
+
+  &smc {
+    status = "okay";
+    pinctrl-0 = <&smc_default>;
+    pinctrl-names = "default";
+
+    is66wv51216dbll@0 {
+      reg = <0>;
+
+      atmel,smc-write-mode = "nwe";
+      atmel,smc-read-mode = "nrd";
+      atmel,smc-setup-timing = <1 1 1 1>;
+      atmel,smc-pulse-timing = <6 6 6 6>;
+      atmel,smc-cycle-timing = <7 7>;
+    };
+  };
+
+  Finally, in order to make the memory available you will need to define new
+  memory device/s in DeviceTree:
+
+  sram1: sram@60000000 {
+      compatible = "zephyr,memory-region", "mmio-sram";
+      device_type = "memory";
+      reg = <0x60000000 DT_SIZE_K(512)>;
+      zephyr,memory-region = "SRAM1";
+  };
+
+compatible: "atmel,sam-smc"
+
+include: [base.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  "#address-cells":
+    required: true
+    const: 1
+
+  "#size-cells":
+    required: true
+    const: 0
+
+  peripheral-id:
+    type: int
+    description: peripheral ID
+    required: true
+
+child-binding:
+  description: |
+    Child device nodes are representing devices connected to the EBI/SMC bus.
+
+  properties:
+    reg:
+      type: int
+      required: true
+      description: |
+        The device's SMC CS number.
+        Valid range: 0 - 3
+
+    atmel,smc-write-mode:
+      type: string
+      required: false
+      default: "nwe"
+      description: |
+        Selects the mode that indicates which signal is used for the write
+        operation. Reset and default value is nwe.
+      enum:
+        - "ncs"
+        - "nwe"
+
+    atmel,smc-read-mode:
+      type: string
+      required: false
+      default: "nrd"
+      description: |
+        Selects the mode that indicates which signal is used for the read
+        operation. Reset and default value is nrd.
+      enum:
+        - "ncs"
+        - "nrd"
+
+    atmel,smc-setup-timing:
+      type: array
+      required: false
+      default: [1, 1, 1, 1]
+      description: |
+        SMC setup timing configuration. Refer to the datasheet for more
+        information. Expected fields, in order, are,
+
+        - NWE
+        - NCS_WR
+        - NRD
+        - NCS_RD
+
+    atmel,smc-pulse-timing:
+      type: array
+      required: false
+      default: [1, 1, 1, 1]
+      description: |
+        SMC pulse timing configuration. Refer to the datasheet for more
+        information. Expected fields, in order, are,
+
+        - NWE
+        - NCS_WR
+        - NRD
+        - NCS_RD
+
+    atmel,smc-cycle-timing:
+      type: array
+      required: false
+      default: [3, 3]
+      description: |
+        SMC cycle timing configuration. Refer to the datasheet for more
+        information. Expected fields, in order, are,
```

Comment:
remove white space.  Also, not sure how `fields` NWE/NRD and an array quite make sense together.

Reply:
I used the array type to create a compact view instead of an entry for each item, and the values are set into the same HW register.

Reply:
The array is fine, I just find the wording confusing.  not sure if `[NWE NRD]` in the description would make it a little clearer.